### PR TITLE
ci: Split up in multiple jobs

### PR DESCRIPTION
**Pull request by andistorm**
## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [ ] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -15,14 +15,29 @@ on:
...
+      - name: Run clang-format
+        uses: everest/everest-ci/github-actions/run-clang-format@v1.0.0
+        with:
+          source-dir: source
+          extensions: hpp, cpp
+  compile:
+    name: Configure and Compile
+    runs-on: ${{ inputs.runner || 'ubuntu-22.04' }}
+    steps:
+      # LTODO: use github actions outputs or custom github actions
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:35]: DO NOT MERGE

File: .github/workflows/build_and_test.yaml [hunk 2]
```diff
@@ -15,14 +15,29 @@ on:
...
       - name: Format branch name for cache key
         run: |
           BRANCH_NAME_FOR_CACHE="${GITHUB_REF_NAME//-/_}"
           echo "branch_name_for_cache=${BRANCH_NAME_FOR_CACHE}" >> "$GITHUB_ENV"
+      # LTODO: use compile-main ??
```
Comment (andistorm) [inline .github/workflows/build_and_test.yaml:40]: DO NOT MERGE

**Comment by barsnick** [comment]
I suggest to mark this "CI" in the final commit message, or PR title, to avoid confusion.

**Comment by andistorm** [comment]
Close in favor of upcoming reusable workflow in everest-ci https://github.com/EVerest/everest-ci/pull/54

**Pull request closed without merging**